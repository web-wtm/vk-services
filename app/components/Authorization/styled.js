import styled from 'styled-components'
import Button from '../Button'

export const Info = styled.div`
    font-size: 22px;
    display: inline-block;
    margin-top: 150px;
    padding: 15px;
    box-shadow: 0 2px 4px rgba(0, 0, 0, 0.3);
    background-color: #fff;

    @media screen and (max-width: 900px) {
        font-size: 16px;
    }

    @media screen and (max-width: 600px) {
        font-size: 12px;
        line-height: 2;
    }
`

export const LogIn = styled.a`
    background-color: #47b475;
    color: #fff;
    text-decoration: none;
    padding: 5px 10px;
    margin-left: 10px;
    text-transform: uppercase;
    transition: all ease 333ms;

    &:hover {
        background-color: #1d4076;
    }
`

export const SaveToken = Button.extend`
    padding: 5px;
    margin-left: 10px;
`