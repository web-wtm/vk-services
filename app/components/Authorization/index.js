import React, { Fragment } from 'react'
import { parse } from 'query-string'
import { Info, LogIn, SaveToken } from './styled'
import Button from '../Button'

const Authorization = (props) => {
    const signInHref = 'https://oauth.vk.com/authorize?client_id=6104841&display=page&redirect_uri=https://vk-services.firebaseapp.com&scope=friends&response_type=token&v=5.65'
    
    return (
        <Info>
            {
                !props.token ? 
                <Fragment>
                    For use all services of app you need to be authorized in VK &#8594;
                    <LogIn href={signInHref}>
                        log in
                    </LogIn>
                </Fragment>
                :
                <Fragment>
                        Great, now you need just &#8594;
                    <SaveToken onClick={props.onClick}>save token</SaveToken>
                </Fragment>
            }
        </Info>
    )
}   

export default Authorization