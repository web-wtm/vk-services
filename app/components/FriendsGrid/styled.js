import styled from 'styled-components'
import UserDeletedThumb from '../../assets/images/user-deleted.png'

export const FriendsContainer = styled.div`
    text-align: center;
    margin-left: 270px;
`

export const User = styled.a`
    display: inline-block;
    margin: 20px;
    color: #333;
    transition: color ease-in 333ms;

    &:hover {
        color: #47b475;
    }
`

export const UserName = styled.p`
    font-family: 'Podkova', serif;
    text-align: center;
    font-weight: bold;
    font-size: 18px;
`

export const UserPhoto = styled.img`
    vertical-align: middle;
    border-radius: 50%;
    margin-right: 10px;
    border: 3px solid #FFF;
`

export const UserDeleted = styled.div`
    background-image: url(${UserDeletedThumb});
    background-size: 100%;
    vertical-align: middle;
    border-radius: 50%;
    margin-right: 10px;
    height: 200px;
    width: 200px;
` 

