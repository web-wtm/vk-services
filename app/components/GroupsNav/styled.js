import styled from 'styled-components'
import Button from '../Button'

export const GroupSearch = styled.div`

`

export const ButtonSearch = Button.extend`
    width: 100%;
`

export const GroupsNavList = styled.ul`
    display: inline-block;
    margin: 0;
    padding: 0;
    width: 100%;
`
export const GroupsNavItem = styled.li`
    display: block;
    cursor: pointer;
    padding: 15px 0;
    margin: 1px 0;
    font-size: 18px;
    font-weight: bold;
    color: #e5e5e5;
    transition: background ease-in 333ms;

    &:first-child {
        margin: 0;
    }

    &:hover {
        background-color: #47b475;
    }
`

export const GroupLogo = styled.img`
    vertical-align: middle;
    width: 30px;
    border-radius: 50%;
    margin: -5px 10px 0;
`