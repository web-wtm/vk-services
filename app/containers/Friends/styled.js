import styled from 'styled-components'
import Button from '../../components/Button'
import Caption from '../../components/Caption'

export const FriendsSearchInfo = Caption.extend`
    margin-left: 270px;

    @media (max-width: 600px) {
        margin-left: 160px;
    }
`

export const ButtonSearch = Button.extend`
    width: 100%;
`