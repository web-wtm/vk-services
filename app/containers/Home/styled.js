import styled from 'styled-components'

export const Container = styled.div`
    min-width: 200px;
    max-width: 1140px;
    margin: 0 auto;
    padding: 0 1%;
`

export const AlreadyAuth = styled.div`
    box-shadow: 0 2px 4px rgba(0,0,0,0.3);  
    background-color: #fff;
    padding: 20px;
    display: inline-block;
    font-size: 28px;
    margin-top: 150px;
    position: relative;

    span {
        background-color: rgba(0, 194, 6, 0.4);
        padding: 6px 10px;
        margin-left: 15px;
        border-radius: 50%;
        font-size: 30px;
        color: #fff;
    }
`