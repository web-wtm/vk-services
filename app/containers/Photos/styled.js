import styled from 'styled-components'

const PhotosStyled = styled.div`
     
`

export default PhotosStyled