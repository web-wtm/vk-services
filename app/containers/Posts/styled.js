import styled from 'styled-components'

const PostsStyled = styled.div`
    position: relative;
`

export default PostsStyled