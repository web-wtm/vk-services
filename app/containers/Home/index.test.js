import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import Authorization from '../../components/Authorization'
import { LogIn, SaveToken } from '../../components/Authorization/styled'

describe('Authorization component', () => {
    const saveFn = jest.fn();
    const component = shallow(<Authorization token={null} onClick={saveFn}/>);

    it('rendered Authorization component', () => {
        const component = renderer.create(<Authorization token={null} onClick={saveFn}/>);
        const tree = component.toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('rendered login link', () => {
        expect(component.find(LogIn)).toHaveLength(1);
    });

    it('rendered save token button', () => {
        component.setProps({token:'test'});
        expect(component.find(SaveToken)).toHaveLength(1);
    });
    
    it('should call function when click save token', () => {
        component.find(SaveToken).simulate('click');
        expect(saveFn).toHaveBeenCalled();
    });
});