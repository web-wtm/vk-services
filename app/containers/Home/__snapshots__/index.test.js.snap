// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Authorization component rendered Authorization component 1`] = `
<div
  className="sc-bwzfXH HGEod"
>
  For use all services of app you need to be authorized in VK â†’
  <a
    className="sc-htpNat cjYSCf"
    href="https://oauth.vk.com/authorize?client_id=6104841&display=page&redirect_uri=https://vk-services.firebaseapp.com&scope=friends&response_type=token&v=5.65"
  >
    log in
  </a>
</div>
`;
